boards/arm/nrf52840_pca10059/nrf52840_pca10059.dts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840.dtsi>
+
+/ {
+	model = "Nordic PCA10059 Dev Kit";
+	compatible = "nordic,pca10059-dk", "nordic,nrf52840-qiaa",
+		     "nordic,nrf52840";
+
+	chosen {
+		zephyr,console = &uart0;
+#ifdef CONFIG_MCUMGR_SMP_UART
+		zephyr,uart-mcumgr = &uart0;
+#endif
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led@0 {
+			gpios = <&gpio0 6 GPIO_INT_ACTIVE_LOW>;
+			label = "Onboard GREEN LED 0";
+		};
+		led1: led@1 {
+			gpios = <&gpio0 8 GPIO_INT_ACTIVE_LOW>;
+			label = "Onboard RED LED 1";
+		};
+		led2: led@2 {
+			gpios = <&gpio1 9 GPIO_INT_ACTIVE_LOW>;
+			label = "Onboard GREEN LED 2";
+		};
+		led3: led@3 {
+			gpios = <&gpio0 12 GPIO_INT_ACTIVE_LOW>;
+			label = "Onboard BLUE LED 3";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		user_button_0: button@0 {
+			label = "Push button switch 0";
+			gpios = <&gpio1 6 GPIO_PUD_PULL_UP>;
+		};
+	};
+};
+
+&gpiote {
+	status ="ok";
+};
+
+&gpio0 {
+	status ="ok";
+};
+
+&gpio1 {
+	status ="ok";
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "ok";
+};
+
+&i2c0 {
+	status = "ok";
+	sda-pin = <26>;
+	scl-pin = <27>;
+};
+
+&i2c1 {
+	status = "ok";
+	sda-pin = <30>;
+	scl-pin = <31>;
+};
+
+&flash0 {
+	/*
+	 * For more information, see:
+	 * http://docs.zephyrproject.org/devices/dts/flash_partitions.html
+	 */
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@e0000 {
+			label = "mcuboot";
+			reg = <0x0000e0000 0x0000c000>;
+		};
+		slot0_partition: partition@1000 {
+			label = "image-0";
+			reg = <0x00001000 0x00006d000>;
+		};
+		slot1_partition: partition@6e800 {
+			label = "image-1";
+			reg = <0x006e000 0x00006d000>;
+		};
+		scratch_partition: partition@ec000 {
```

Comment:
It looks there is some space left after slot-1 and before mcu-boot partition

Reply:
I see that this is taken by storage partition, but this is conditional.

Reply:
Slot 1 ends at 0xdb000. MCUBoot starts at 0xe0000. There is a gap between them if there is no storage partition.

Reply:
Correct, that gap can be reduced to one page but it cannot disappear unless the storage partition is placed after the bootloader. That's because the offset of the bootloader is fixed (@ 0xE0000) and that's an even flash page; slot0 begins at 0x1000, 0xE0000 - 0x1000 = 0xDF000 which is odd.